go/vt/srvtopo/server.go

```diff
@@ -47,3 +47,71 @@ type Server interface {
 	// a new value or an error occurs.
 	WatchSrvVSchema(ctx context.Context, cell string, callback func(*vschemapb.SrvVSchema, error))
 }
+
+// NewKeyspaceFilteringServer constructs a new server based on the provided
+// implementation that prevents the specified keyspaces from being exposed
+// to consumers of the new Server.
+func NewKeyspaceFilteringServer(underlying Server, selectedKeyspaces []string) Server {
```

Comment:
IMO this should be pulled into a separate `keyspace_filtering_server.go` implementation file.

Reply:
✅